### ISSUE TITLE
Share game board

### ISSUE THREAD

**Issue by quozl** [sugarlabs/cookie-search-activity#18]
When the activity is shared, the two users play different maps.  They should play in the same map.

Summary: Add collaboration to Cookie Search activity

Many Sugar activities can share their state among a group of children using the same wireless network.

Cookie Search doesn't currently do this, though it does import some of the required library. It could be a bug, or it could be unfinished.

Our CollabWrapper library can be used, and it includes a test activity.

You may need to Port to Python 3 to get started, if nobody else has done that yet.

**Comment by sanjaymaniam** [comment]
@quozl I'd like to try fixing this, but I presently do not have access to another machine. I'm thinking of using two virtual instances of Sugar Desktop on a host. Is this an advisable way to test collaborate or do I need to arrange for another machine before I get started?

**Comment by quozl** [comment]
Thanks for asking.  Collaboration can be tested with two or three virtual machines.  Install Sugar on each in turn.

Don't clone the virtual machines unless you either (a) clone before running Sugar the first time, or (b) remove all the Sugar user identity files in the home directory.  Otherwise collaboration will fail.

The virtual machines must be configured to share a virtual network.  I can't be precise about that because there are lots of different ways to run virtual machines.  But if they can ping each other by mDNS resolution after changing hostname to be unique, then the virtual network is operating correctly;

```bash
ping -c 10 hostname.local
```

Test your configuration against collaboration using the Chat activity.  See [documentation](https://help.sugarlabs.org/en/chat.html).

Also look at the CollabWrapper test activity.

Feel free to search for and maintain, or write up a _How to test collaboration_ section of sugar-docs repository.

**Comment by quozl** [comment]
Collaboration code in this activity appears to be a copy and paste from another activity during development, so adding collaboration will take much thought.  Alternatively, it could be removed.

**Comment by sanjaymaniam** [comment]
Yes. It was very similar to the Flip Activity. However, I couldn't wrap my head around the need for collaborate in Cookie Search. I'm _not_ working on this for that reason. If I were to start over, I'd play a bunch of multiplayer minesweeper first.

**Comment by quozl** [comment]
Thanks.  Our misguidance came from "oh look, it imports Telepathy, we have to port it and make sure collaboration works."
